File: webapp/src/main/java/jm/controller/WorkspaceRestController.java
package jm.controller;

import jm.Workspace;
import jm.WorkspaceService;
import org.hibernate.jdbc.Work;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/workspace")
public class WorkspaceRestController {

    private WorkspaceService workspaceService;

    @Autowired
    public void setWorkspaceService(WorkspaceService workspaceService) {
        this.workspaceService = workspaceService;
    }

    @RequestMapping(value = "/all", method = RequestMethod.GET)
    public ResponseEntity<List<Workspace>> allWorkspaces() {
        List<Workspace> workspaces = null;
        workspaces = workspaceService.gelAllWorkspaces();
        return new ResponseEntity<>(workspaces, HttpStatus.OK);
    }

    @RequestMapping(value = "/delete", method = RequestMethod.DELETE)
    public ResponseEntity<?> deleteWorkspace(@RequestParam("id") Long id) {
        workspaceService.deleteWorkspace(id);
        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
    }

    @RequestMapping(value = "/create", method = RequestMethod.POST)
    public ResponseEntity<?> createWorkspace(@RequestBody Workspace workspace) {
        workspaceService.createWorkspace(workspace);
        return new ResponseEntity<>(workspace, HttpStatus.OK);
    }

    @RequestMapping(value = "/update", method = RequestMethod.PUT)
    public ResponseEntity<?> updateWorkspace(@RequestBody Workspace workspace) {
        workspaceService.updateWorkspace(workspace);
        return new ResponseEntity<>(workspace, HttpStatus.OK);
    }

    @RequestMapping(value = "/getById", method = RequestMethod.GET)
    public ResponseEntity<Workspace> getWorkspaceById(@RequestParam("id") Long id) {
        Workspace workspace = workspaceService.getWorkspaceById(id);
        return new ResponseEntity<>(workspace, HttpStatus.OK);
    }

}

File: webapp/src/main/resources/static/js/ajax/workspaceRestController.js
export const getWorkspaceById = (id) => {
    let result = null;
    $.ajax({
        type: 'get',
        async: false,
        url: "/rest/api/workspaces/workspace/" + id
    }).done(function (data) {
        result = data;
    });
    return result;
};

export const createWorkspace = (workspace) => {
    let result = null;
    const jsonWorkspace = JSON.stringify(workspace);
    $.ajax({
        type: 'post',
        url: "/rest/api/workspaces/create",
        async: false,
        data: jsonWorkspace,
        contentType: "application/json"
    }).done(function (data) {
        result = data;
    });
    return result;
};

export const updateWorkspace = (workspace) => {
    const jsonWorkspace = JSON.stringify(workspace);
    $.ajax({
        type: 'put',
        url: "/rest/api/workspaces/update",
        async: false,
        data: jsonWorkspace,
        contentType: "application/json"
    });
};

export const deleteWorkspace = (id) => {
    $.ajax({
        type: 'delete',
        url: "/rest/api/workspaces/delete/" + id,
        async: false
    });
};

export const getAllWorkspaces = () => {
    let workspaces = null;
    $.ajax({
        url: '/rest/api/workspaces/',
        async: false,
        type: 'get',
        success: function (response) {
            workspaces = response;
        }
    });
    return workspaces;
};

